Extension/test/unit/eventing.test.ts

```diff
@@ -0,0 +1,255 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+import { ok, strictEqual } from 'assert';
+import { beforeEach, describe, it } from 'mocha';
+import { Async } from '../../src/Utility/Async/constructor';
+import { sleep } from '../../src/Utility/Async/sleep';
+import { Descriptors } from '../../src/Utility/Eventing/descriptor';
+import { notify, notifyNow, on, reset, subscribe } from '../../src/Utility/Eventing/dispatcher';
+import { Emitter } from '../../src/Utility/Eventing/emitter';
+import { Cancelled, Continue, EventData } from '../../src/Utility/Eventing/interfaces';
+import { is } from '../../src/Utility/System/guards';
+
+export class SomeBase extends Emitter {
+
+}
+
+class Something extends SomeBase {
+    constructor() {
+        super();
+        this.descriptors.add('server', 'mysrv');
+    }
+
+    // emitter declarations!

```

Comment:
Is this comment supposed to end with a `!`?

Reply:
I guess I was excited.